modules/nf-core/rgi/main/tests/main.nf.test

```diff
@@ -0,0 +1,95 @@
+nextflow_process {
+
+    name "Test Process RGI_MAIN"
+    script "../main.nf"
+    process "RGI_MAIN"
+
+    tag "modules"
+    tag "modules_nfcore"
+    tag "rgi"
+    tag "rgi/main"
+    tag "rgi/cardannotation"
+    tag "untar"
+
+    setup {
+        run("UNTAR") {
+            script "modules/nf-core/untar/main.nf"
+            process {
+            """
+            file('https://card.mcmaster.ca/latest/data', checkIfExists: true).copyTo('data.tar.gz')
+
+            input[0] = [
+                [ ],
+                file("data.tar.gz")
+            ]
+            """
+            }
+        }
+
+        run("RGI_CARDANNOTATION") {
+            script "modules/nf-core/rgi/cardannotation"
+            process {
+                """
+                input[0] = UNTAR.out.untar.map{ it[1] }
+                """
+            }
+
+        }
+    }
+
+
+    test("rgi/main - haemophilus_influenzae - genome_fna_gz") {
+
+        when {
+            process {
+                """
+                input[0] = [                                                                                                                                                                   
+                    [ id:'test', single_end:false ], // meta map                                                                                                                            
+                    file(params.test_data['haemophilus_influenzae']['genome']['genome_fna_gz'], checkIfExists: true)                                                                        
+                ]
+                input[1] = RGI_CARDANNOTATION.out.db
+                input[2] = []
```

Comment:
Can you add a test with the wildcard database? Or is that tricky because of the database?

Reply:
Wildcard archive file is 1.6 GB... Shall I still? It might break GHA again. :fire:

Reply:
Ooof, I think I would say yes try (even though slow), given the complexity of the command

Reply:
Fixed all comments/suggestions but this. Locally I created the new module `rgi/wildcardannotation`, will prbly push the final update it tonight :+1:

Reply:
Do you think it'd be okay to add the `rgi_wildcardannotation` (and your requested test of that in `rgi_main`) separately later? I will do it but it probably takes 1 whole day, it's more complicated than the card-only database and needs testing which slows me down. If you find it necessary for this PR, it's okay, can prbly only do it next week then.

Reply:
My question would be:
1. Merge this now + add `rgi_wildcardannotation` later or
2. Have it all in this PR but a bit slower

Reply:
Number 1 as it's a separate module